engine/apps/user_management/models/user.py

```diff
@@ -85,6 +74,36 @@ def filter(self, *args, **kwargs):
     def filter_with_deleted(self, *args, **kwargs):
         return super().filter(*args, **kwargs)
 
+    def filter_by_permission(self, permission: LegacyAccessControlCompatiblePermission,
+                             organization: "Organization", *args, **kwargs):
+        """
+        This method builds a filter query that is compatible with RBAC as well as legacy "basic" role based
+        authorization. If a permission is provided we simply do a regex search where the permission column
+        contains the permission value (need to use regex because the JSON contains method is not supported by sqlite).
+        Additionally, we check whether the user has either the `grafana-oncall-app` OR `grafana-irm-app`
+        prefixed-version of the permission.
+
+        If RBAC is not supported for the org, we make the assumption that we are looking for any users with AT LEAST
+        the fallback role. Ex: if the fallback role were editor than we would get editors and admins.
+        """
+        if organization.is_rbac_permissions_enabled:
+            # https://stackoverflow.com/a/50251879
+            if settings.DATABASE_TYPE == settings.DATABASE_TYPES.SQLITE3:
+                # contains is not supported on sqlite
+                def _build_q_object(value: str) -> Q:
+                    return Q(permissions__regex=re.escape(value))
+
+                query = _build_q_object(permission.value_oncall_app) | _build_q_object(permission.value_irm_app)
+            else:
+                def _build_q_object(value: str) -> Q:
+                    return Q(permissions__contains=User.construct_permissions_from_actions([value]))
+
+                query = _build_q_object(permission.value_oncall_app) | _build_q_object(permission.value_irm_app)
+        else:
+            query = Q(role__lte=permission.fallback_role.value)
+
+        return self.filter(query, organization=organization, *args, **kwargs)
```

Comment:
this has been moved from `User.build_permissions_query` (and slightly refactored). It's much simpler to do something like:
```python3
queryset = User.objects.filter_by_permission(RBACPermission.Permissions.NOTIFICATIONS_READ, organization)
```
vs:

```python3
queryset = User.objects.filter(
    organization=organization,
    **User.build_permissions_query(RBACPermission.Permissions.NOTIFICATIONS_READ, organization),
)
```